core/include/utils/moving_average.h

```diff
@@ -0,0 +1,19 @@
+#include <vector>
+
+class MovingAverage {
```

Comment:
Would it make sense to move this class to the math_utils.h file? I'd also like to see a top level comment with a brief description of what the class does, similar to what the functions in math_utils.h have

Reply:
Top comment added. As for including it in math_utils, I don't know enough about header files to know if that would work or if it would be messy.